### PR TITLE
Allow using test harness in a single unit test

### PR DESCRIPTION
**Pull request by piotrswigon**
Pulls out setup and cleanup code of the `BasteTestHarness` in order to reuse it in `runHarnessTest` utility method for setting up and cleaning up a harness for a single unit test, instead of setting up a rule for all tests in the file.

This is useful if one would like to group tests with multiple different particles in the same file, like I do in EvaluatorParticleTest in https://github.com/PolymerLabs/arcs/pull/6222

### CODE REVIEW

File: java/arcs/sdk/testing/BaseTestHarness.kt [hunk 1]
```diff
@@ -91,53 +110,59 @@ open class BaseTestHarness<P : Particle>(
...
+                        spec.copy(mode = HandleMode.ReadWrite),
+                        storageKey,
+                        immediateSync = false
+                    )
                 }
             }
+            run()
+            runBlocking {
+                scheduler.waitForIdle()
+                handleManager.close()
```
Comment (cromwellian) [inline java/arcs/sdk/testing/BaseTestHarness.kt:162]: shouldn't the handleManager.close() be in the finally below?

**Review by cromwellian — APPROVED** [review]

File: java/arcs/sdk/testing/BaseTestHarness.kt (hunk 1 above)
Comment (piotrswigon) [reply java/arcs/sdk/testing/BaseTestHarness.kt]: I think so. Moved.

**Pull request closed without merging**